lib/Dialect/HW/HWTypes.cpp

```diff
@@ -270,6 +270,14 @@ Optional<unsigned> StructType::getFieldIndex(mlir::StringRef fieldName) {
   return {};
 }
 
+Optional<unsigned> StructType::getFieldIndex(mlir::StringAttr fieldName) {
+  ArrayRef<hw::StructType::FieldInfo> elems = getElements();
+  for (unsigned idx = 0, numElems = elems.size(); idx < numElems; ++idx)
```

Comment:
size_t, though if we ever have that many fields, we have other problems.

Reply:
Done in 9ca2477f